### PR TITLE
Update columns in covid_hosp: facility, state daily, state timeseries

### LINKED ISSUE #720: Update available columns in covid_hosp datasets
There have been quite a few columns added to the HHS datasets on patient impact and hospital capacity since we last updated our code, including:

* age stratifications added April 30
* covid deaths added June 30
* drug courses on hand added September 13
* influenza admissions, ICU prevalence, and deaths, added September 17

The above are the additions for state timeseries; state daily and facility weekly may have a different set.

The code for this dataset is in [`acquisition/covid_hosp`](https://github.com/cmu-delphi/delphi-epidata/tree/dev/src/acquisition/covid_hosp). There are three variants: state timeseries, state daily, and facility (weekly).

Updating that code requires the following steps:
1. Look up the CSV column headers for all new columns on the appropriate website ([state timeseries is here](https://healthdata.gov/Hospital/COVID-19-Reported-Patient-Impact-and-Hospital-Capa/g62h-syeh); you should be able to find links for the others in the [Epidata docs on non-covidcast endpoints](https://cmu-delphi.github.io/delphi-epidata/api/README.html))
2. Add the new columns to the database definitions in [covid_hosp.sql](https://github.com/cmu-delphi/delphi-epidata/blob/dev/src/ddl/covid_hosp.sql) (possibly using an alias name to reduce column name length in the db)
3. List the mapping between CSV column header, DB column name, and data type in each dataset's [database.py (example: state timeseries)](https://github.com/cmu-delphi/delphi-epidata/blob/dev/src/acquisition/covid_hosp/state_timeseries/database.py)
4. Update or extend the tests ([unit](https://github.com/cmu-delphi/delphi-epidata/tree/dev/tests/acquisition/covid_hosp); [integration](https://github.com/cmu-delphi/delphi-epidata/tree/dev/integrations/acquisition/covid_hosp)) with new sample data containing the new columns
5. Write a database migration script (in SQL) to add the new columns to the existing database. We haven't historically done this in a reviewable fashion, but we should really start. Make a new directory `src/ddl/migrations/` to hold this script, and include the acquisition module (`covid_hosp`) in the name, as well as the starting and ending minor versions (we're currently in v0.2; this change includes a database migration so it will push us into v0.3. You can always see the current complete version on the [repository releases page](https://github.com/cmu-delphi/delphi-epidata/releases))
6. Open a PR and tag a reviewer.
7. Once your PR is approved, merge it, then ping Katie to schedule a minor-version release. Release should occur after the day's covid_hosp updates run in Automation, which are scheduled for 10am but have been delayed lately due to the covidcast meta cache running time problems. Avoid releasing on Fridays for everyone's health.
8. After release and before the next day's covid_hosp update jobs run in Automation, run the migration.

### PR DESCRIPTION
**Pull request by krivard**
closes #720 

**Prerequisites**:

- [x] Unless it is a documentation hotfix it should be merged against the `dev` branch
- [x] Branch is up-to-date with the branch to be merged with, i.e. `dev`
- [x] Build is successful
- [x] Code is cleaned up and formatted

### Summary

State:
* Add new columns to `covid_hosp_state_timeseries` table with SQL-compatible name length and spelling
* Add database migration helper
* Add corresponding entries to `database.py` for state daily and state timeseries acquisition pipelines
* Add corresponding entries to `covid_hosp_state_timeseries.py` server endpoint

Facility:
* Add new columns to `covid_hosp_facility` table with SQL-compatible name length and spelling
* Add database migration helper
* Add corresponding entries to `database.py` for facility acquisition pipeline
* Add corresponding entries to `covid_hosp_facility.py` server endpoint

Unit and integration tests updated with new constants and sample data.

### CODE REVIEW

File: integrations/acquisition/covid_hosp/facility/test_scenarios.py [hunk 1]
```diff
@@ -54,7 +55,7 @@ def test_acquire_dataset(self):
     with self.subTest(name='no data yet'):
       response = Epidata.covid_hosp_facility(
           '450822', Epidata.range(20200101, 20210101))
-      self.assertEqual(response['result'], -2)
+      self.assertEqual(response['result'], -2, response)
```
[the remark this answers is not in the document]
Comment (krivard) [reply integrations/acquisition/covid_hosp/facility/test_scenarios.py:58]: the third argument of `self.assert*()` is the message it prints on failure. so if it fails, it prints out the actual response, including the `message`.  makes debugging way easier than just getting "-1 != -2"

[the remark this answers is not in the document]
Comment (chinandrew) [reply integrations/acquisition/covid_hosp/facility/test_scenarios.py:58]: oh cool didn't know that, thanks

File: src/acquisition/covid_hosp/common/database.py [hunk 2]
```diff
@@ -157,6 +157,10 @@ def insert_dataset(self, publication_date, dataframe):
     sql = f'INSERT INTO `{self.table_name}` (`id`, `{self.publication_col_name}`, {columns}) ' \
           f'VALUES ({value_placeholders})'
     id_and_publication_date = (0, publication_date)
+    NEWLINE="\n"
+    for name, _, _ in self.columns_and_types:
+      if name not in dataframe.columns:
+        raise Exception(f"{name} not in dataframe columns:\n{NEWLINE.join(sorted(dataframe.columns))}")
```
Comment (krivard) [inline src/acquisition/covid_hosp/common/database.py:163]: how will this behave if we run `fill_covid_hosp_skipped_issue.py` with a file from before all the columns were added? have the hhs files all been back-updated with all the columns?

Comment (chinandrew) [reply src/acquisition/covid_hosp/common/database.py:163]: Looking at one of the [old archived files ](https://healthdata.gov/dataset/COVID-19-Reported-Patient-Impact-and-Hospital-Capa/qqte-vkut)from March, it doens't have the new columns, so presumably it'd just error?

Comment (krivard) [reply src/acquisition/covid_hosp/common/database.py]: that should do it?

File: src/ddl/migrations/covid_hosp_facility_v0.2-v0.3.sql [hunk 3]
```diff
@@ -0,0 +1,20 @@
+ALTER TABLE covid_hosp_facility ADD
```
Comment (krivard) [inline src/ddl/migrations/covid_hosp_facility_v0.2-v0.3.sql:1]:
```suggestion
ALTER TABLE covid_hosp_facility ADD (
```

File: src/ddl/migrations/covid_hosp_facility_v0.2-v0.3.sql [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+    `previous_day_admission_adult_covid_confirmed_7_day_coverage` INT,
+    `previous_day_admission_adult_covid_suspected_7_day_coverage` INT,
+    `previous_day_admission_pediatric_covid_confirmed_7_day_coverage` INT,
+    `previous_day_admission_pediatric_covid_suspected_7_day_coverage` INT,
+    `previous_week_patients_covid_vaccinated_doses_all_7_day` INT,
+    `previous_week_patients_covid_vaccinated_doses_all_7_day_sum` INT,
+    `previous_week_patients_covid_vaccinated_doses_one_7_day` INT,
+    `previous_week_patients_covid_vaccinated_doses_one_7_day_sum` INT,
+    `previous_week_personnel_covid_vaccinated_doses_administered_7_day` INT,
+    `previous_week_personnel_covid_vaccinated_doses_administered_7_day_sum` INT,
```
Comment (krivard) [inline src/ddl/migrations/covid_hosp_facility_v0.2-v0.3.sql:14]:
```suggestion
    `previous_week_personnel_covid_vaccd_doses_administered_7_day` INT,
    `previous_week_personnel_covid_vaccd_doses_administered_7_day_sum` INT,
```

File: src/ddl/migrations/covid_hosp_facility_v0.2-v0.3.sql [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+    `total_personnel_covid_vaccinated_doses_all_7_day` INT,
+    `total_personnel_covid_vaccinated_doses_all_7_day_sum` INT,
+    `total_personnel_covid_vaccinated_doses_none_7_day` INT,
+    `total_personnel_covid_vaccinated_doses_none_7_day_sum` INT,
+    `total_personnel_covid_vaccinated_doses_one_7_day` INT,
+    `total_personnel_covid_vaccinated_doses_one_7_day_sum` INT;
```
Comment (krivard) [inline src/ddl/migrations/covid_hosp_facility_v0.2-v0.3.sql:20]:
```suggestion
    `total_personnel_covid_vaccinated_doses_one_7_day_sum` INT);
    
```

File: src/ddl/migrations/covid_hosp_state_v0.2-v0.3.sql [hunk 6]
```diff
@@ -0,0 +1,58 @@
+ALTER TABLE covid_hosp_state_timeseries ADD
```
Comment (krivard) [inline src/ddl/migrations/covid_hosp_state_v0.2-v0.3.sql:1]:
```suggestion
ALTER TABLE covid_hosp_state_timeseries ADD (
```

File: src/ddl/migrations/covid_hosp_state_v0.2-v0.3.sql [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+ `previous_day_admission_influenza_confirmed` INT,
+ `previous_day_admission_influenza_confirmed_coverage` INT,
+ `previous_day_deaths_covid_and_influenza` INT,
+ `previous_day_deaths_covid_and_influenza_coverage` INT,
+ `previous_day_deaths_influenza` INT,
+ `previous_day_deaths_influenza_coverage` INT,
+ `total_patients_hospitalized_confirmed_influenza` INT,
+ `total_patients_hospitalized_confirmed_influenza_covid` INT,
+ `total_patients_hospitalized_confirmed_influenza_covid_coverage` INT,
+ `total_patients_hospitalized_confirmed_influenza_coverage` INT;
```
Comment (krivard) [inline src/ddl/migrations/covid_hosp_state_v0.2-v0.3.sql:58]:
```suggestion
 `total_patients_hospitalized_confirmed_influenza_coverage` INT);
```

**Pull request merged**